tests/test_edgeql_calls.py

```diff
@@ -1141,3 +1141,137 @@ class TestEdgeQLFuncCalls(tb.QueryTestCase):
                 [1, 2],
             ]
         ])
+
+    @test.xfail('''
+        The operator IS is normally static, so special care needs to
+        be taken when it's used inside polymorphic functions.
+    ''')
+    async def test_edgeql_calls_33(self):
+        await self.con.execute('''
+            CREATE FUNCTION test::call33(
+                a: anytype
+            ) -> bool
+                FROM EdgeQL $$
+                    SELECT a IS float32
+                $$;
+
+            CREATE TYPE test::A33;
+
+            INSERT A33;
+        ''')
+
+        try:
+            await self.assert_query_result(r'''
+                SELECT test::call33('foo');
+                SELECT test::call33([3]);
+                SELECT test::call33([<float32>3]);
+
+                SELECT test::call33(<int16>2);
+                SELECT test::call33(<int32>2);
+                SELECT test::call33(<int64>2);
+                SELECT test::call33(<float32>2);
+                SELECT test::call33(<float64>2);
+                SELECT test::call33(<decimal>2);
+
+                SELECT test::call33(A);
+            ''', [
+                [False],
+                [False],
+                [False],
+
+                [False],
+                [False],
+                [False],
+                [True],
+                [False],
+                [False],
+
+                [False],
+            ])
+
+        finally:
+            await self.con.execute('''
+                DELETE A33;
+
+                DROP TYPE test::A33;
+
+                DROP FUNCTION test::call33(
+                    a: anyreal
+                );
+            ''')
+
+    @test.xfail('''
+        All of the results are False, whereas some are expected to be
+        True because the operator '^' should be polymorphic.
+    ''')
```

Comment:
?

Testing the return type of `^` operator can be done without putting it into a polymorphic function.  The purpose of this test is to show that `float32 ^ float32` is `float32`, right?  AFAIK you can fix this by adding a specialized `^` for `float32` and cast the result there.

Reply:
Dropping the test because polymorphic functions will not be supported in user-space.